### ISSUE TITLE
Comet and Transformers

### ISSUE THREAD

**Issue by f3n1Xx** [comet-ml/issue-tracking#434]
## Before Asking:   
- [x] I have searched the Issue Tracker.   
- [x] I have searched the Documentation.    

## What is your question related to?
- [x] Comet Python SDK
- [ ] Comet UI
- [x] Third Party Integrations (Huggingface, TensorboardX, Pytorch Lightning etc.)  

## What is your question?

I am running comet with huggingface transformers. So my issue is the weird way the config is setup. Let me break it down so that everyone is on same page.

1. I initially started setting up comet with report_to = "comet-ml" (which invokes the CometCallback in their library) in TrainingArguments in their TrainerAPI. But the issue was i cannot access log_confusion matrix of experiment object anymore. So there were two seperate experiments being created. So i removed it and created experiment object outside so i can manipulate accordingly.
2. Now the issue is, i am under the impression that steps being recorded for loss vs step chart is incorrect, where is this loss and step being logged from? i didn't set it anywhere. I want to be clear what values i am logging so that i can make accurate assumptions. 
3. So i went back to CometCallback  now i cannot create confusion matrices, nor i am able to do hyperparameter optimization.  So i am stuck creating experiment object explicitly rather letting transformers library auto log the variables.

Now i would achieve my goals if:
1. there is a way to know which values are being logged wrt loss and steps.
2. any way to integrate comet with tranformers better so that i can enable logging of confusion matrices and performing Optimizer properly.

## Code

Nothing specific.

## What have you tried?
Mentioned above.
